### PR TITLE
Move Participant storage to sqlite database

### PR DESCRIPTION
**Pull request by erikwiffin**
Also moves fhir and oauth URLs into the provider object.

### CODE REVIEW

File: providers.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
     {
         "name": "Timothy Garrick",
         "city": "Boston",
-        "state": "Massachusetts"
+        "state": "Massachusetts",
+        "fhir_url": "http://52.39.26.206:9000/api/fhir",
+        "oauth_url": "http://52.39.26.206:9000/api/oauth",
```
Comment (jmandel) [inline providers.json]: In theory fhir_url is the only thing that should be stored here - the others can be derived from the FHIR /metadata endpoint

File: researchapp/services/oauth.py [hunk 2]
```diff
@@ -35,7 +32,7 @@ def refresh(token):
     client_auth = requests.auth.HTTPBasicAuth('app-demo',
                                               'demo-secret-s4s')
 
-    response = requests.post(TOKEN_URL,
+    response = requests.post(provider['oauth_url'] + '/token',
```
Comment (jmandel) [inline researchapp/services/oauth.py]: We shouldn't assume that the token URL can be constructed this way - the right way to learn it is via /metadata endpoint from the FHIR server

File: researchapp/templates/consent.jinja2 [hunk 3]
```diff
@@ -6,20 +6,18 @@
...
 {% block body %}
 <div>
-    <form action="http://52.39.26.206:9000/authorize/" id="authorize-form">
-        <input type="hidden" name="client_id" value="{{ oauth_config.client_id }}" />
+    <form action="{{ provider.authorize_url }}" id="authorize-form">
+        <input type="hidden" name="client_id" value="app-demo" />
         <input type="hidden" name="response_type" value="code" />
         <input type="hidden" name="scope" value="patient/*.read offline_access" />
         <input type="hidden" name="redirect_uri" value="http://52.39.26.206:6543/authorized" />
         <input type="hidden" name="state" value="123" />
```
Comment (jmandel) [inline researchapp/templates/consent.jinja2]: Eventually this needs to be a high entropy random value that the server component stores (and verifies after redirection  back to the app).

File: providers.json (hunk 1 above)
Comment (erikwiffin) [inline providers.json]: Nice.

File: researchapp/services/oauth.py (hunk 2 above)
Comment (erikwiffin) [inline researchapp/services/oauth.py]: Fixed.

File: researchapp/templates/consent.jinja2 (hunk 3 above)
Comment (erikwiffin) [inline researchapp/templates/consent.jinja2]: I replaced this with the CSRF functionality in pyramid. For this workflow, they behave the same way, but let me know if state generally behaves differently, and treating it like a CSRF token is not appropriate.

Comment (jmandel) [inline researchapp/templates/consent.jinja2]: A best practice is to ensure that `state` is never re-used -- and generally CSRF tokens are fixed for an entire user session (or even longer). So we should generate state separately from CSRF (e.g. could calculate a HMAC over a fixed CSRF token + an incrementing integer, and use that as state).

Comment (erikwiffin) [inline researchapp/templates/consent.jinja2]: If that's the case, it's probably better to not confuse concepts. I'll just use a uuid v4 stored in the session.

Comment (jmandel) [inline researchapp/templates/consent.jinja2]: Yep -- and invalidate/rotate each time a launch occurs.

**Comment by erikwiffin** [comment]
Responded to comments, rebased, should be good to merge now.

**Comment by jmandel** [comment]
Looks good -- thanks!

**Pull request merged**